Core/MIPS/IR/IRInst.cpp

```diff
@@ -0,0 +1,589 @@
+	{ IROp::ShrImm, "ShrImm", "GGI" },
+	{ IROp::SarImm, "SarImm", "GGI" },
+	{ IROp::RorImm, "RorImm", "GGI" },
+	{ IROp::Slt, "Slt", "GGG" },
+	{ IROp::SltConst, "SltConst", "GGC" },
+	{ IROp::SltU, "SltU", "GGG" },
+	{ IROp::SltUConst, "SltUConst", "GGC" },
+	{ IROp::Clz, "Clz", "GG" },
+	{ IROp::MovZ, "MovZ", "GGG" },
+	{ IROp::MovNZ, "MovNZ", "GGG" },
+	{ IROp::Max, "Max", "GGG" },
+	{ IROp::Min, "Min", "GGG" },
+	{ IROp::BSwap16, "BSwap16", "GG" },
+	{ IROp::BSwap32, "BSwap32", "GG" },
+	{ IROp::Mult, "Mult", "_GG" },
+	{ IROp::MultU, "MultU", "_GG" },
+	{ IROp::Madd, "Madd", "_GG" },
+	{ IROp::MaddU, "MaddU", "_GG" },
+	{ IROp::Msub, "Msub", "_GG" },
+	{ IROp::MsubU, "MsubU", "_GG" },
+	{ IROp::Div, "Div", "_GG" },
+	{ IROp::DivU, "DivU", "_GG" },
+	{ IROp::MtLo, "MtLo", "_G" },
+	{ IROp::MtHi, "MtHi", "_G" },
+	{ IROp::MfLo, "MfLo", "G" },
+	{ IROp::MfHi, "MfHi", "G" },
+	{ IROp::Ext8to32, "Ext8to32", "GG" },
+	{ IROp::Ext16to32, "Ext16to32", "GG" },
+	{ IROp::Load8, "Load8", "GGC" },
+	{ IROp::Load8Ext, "Load8", "GGC" },
+	{ IROp::Load16, "Load16", "GGC" },
+	{ IROp::Load16Ext, "Load16Ext", "GGC" },
+	{ IROp::Load32, "Load32", "GGC" },
+	{ IROp::LoadFloat, "LoadFloat", "FGC" },
+	{ IROp::Store8, "Store8", "GGC" },
+	{ IROp::Store16, "Store16", "GGC" },
+	{ IROp::Store32, "Store32", "GGC" },
+	{ IROp::StoreFloat, "StoreFloat", "FGC" },
+	{ IROp::FAdd, "FAdd", "FFF" },
+	{ IROp::FSub, "FSub", "FFF" },
+	{ IROp::FMul, "FMul", "FFF" },
+	{ IROp::FDiv, "FDiv", "FFF" },
+	{ IROp::FMov, "FMov", "FF" },
+	{ IROp::FSqrt, "FSqrt", "FF" },
+	{ IROp::FNeg, "FNeg", "FF" },
+	{ IROp::FAbs, "FAbs", "FF" },
+	{ IROp::FRound, "FRound", "FF" },
+	{ IROp::FTrunc, "FTrunc", "FF" },
+	{ IROp::FCeil, "FCeil", "FF" },
+	{ IROp::FFloor, "FFloor", "FF" },
+	{ IROp::FCvtWS, "FCvtWS", "FF" },
+	{ IROp::FCvtSW, "FCvtSW", "FF" },
+	{ IROp::FMovFromGPR, "FMovFromGPR", "FG" },
+	{ IROp::FMovToGPR, "FMovToGPR", "GF" },
+	{ IROp::FpCondToReg, "FpCondToReg", "G" },
+	{ IROp::VfpuCtrlToReg, "VfpuCtrlToReg", "GI" },
+	{ IROp::SetCtrlVFPU, "SetCtrlVFPU", "TC" },
+	{ IROp::Interpret, "Interpret", "_C" },
+	{ IROp::Downcount, "Downcount", "_II" },
+	{ IROp::ExitToConst, "Exit", "C" },
+	{ IROp::ExitToConstIfEq, "ExitIfEq", "CGG" },
+	{ IROp::ExitToConstIfNeq, "ExitIfNeq", "CGG" },
+	{ IROp::ExitToConstIfGtZ, "ExitIfGtZ", "CG" },
+	{ IROp::ExitToConstIfGeZ, "ExitIfGeZ", "CG" },
+	{ IROp::ExitToConstIfLeZ, "ExitIfLeZ", "CG" },
+	{ IROp::ExitToConstIfLtZ, "ExitIfLtZ", "CG" },
+	{ IROp::ExitToReg, "ExitToReg", "G" },
+	{ IROp::Syscall, "Syscall", "_C" },
+	{ IROp::Break, "Break", ""},
+	{ IROp::SetPC, "SetPC", "_G" },
+	{ IROp::SetPCConst, "SetPC", "_C" },
+	{ IROp::CallReplacement, "CallRepl", "_C"},
+};
+
+const IRMeta *metaIndex[256];
+
+void InitIR() {
+	for (size_t i = 0; i < ARRAY_SIZE(irMeta); i++) {
+		metaIndex[(int)irMeta[i].op] = &irMeta[i];
+	}
+}
+
+u32 IRInterpret(MIPSState *mips, const IRInst *inst, const u32 *constPool, int count) {
```

Comment:
Assuming we add VFPU, etc. this will get a lot longer.  What do you think about having it in a dedicated file?

-[Unknown]